wdl-lint/src/rules/section_order.rs

```diff
@@ -0,0 +1,195 @@
+//! A lint rule for section ordering.
+
+use wdl_ast::v1::TaskDefinition;
+use wdl_ast::v1::TaskItem;
+use wdl_ast::v1::WorkflowDefinition;
+use wdl_ast::v1::WorkflowItem;
+use wdl_ast::AstNode;
+use wdl_ast::AstToken;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::ToSpan;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the section ordering rule.
+const ID: &str = "SectionOrdering";
+
+/// Creates a workflow section order diagnostic.
+fn workflow_section_order(span: Span, name: &str, problem_span: Span) -> Diagnostic {
+    Diagnostic::note(format!("sections are not in order for workflow {}", name))
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_label("first problematic section", problem_span)
```

Comment:
For labels, I like to follow what rustc does and use language such as "this" or "here".

```suggestion
        .with_label("this section is out of order", problem_span)
```